ray-operator/controllers/ray/raycluster_controller_fake_test.go

```diff
@@ -922,15 +769,6 @@ func TestReconcile_PodCrash_Diff0_WorkersToDelete_OK(t *testing.T) {
 	assert.Nil(t, err, "Fail to get pod list")
 	assert.Equal(t, oldNumWorkerPods+numHeadPods, len(podList.Items), "Init pod list len is wrong")
 
-	// Simulate 2 Pod fails. Because the workersToDelete also contains pod3 and pod4, the controller will
-	// delete these two Pods. After the deletion, the number of worker Pods should be 3.
-	podList.Items[3].Status.Phase = corev1.PodFailed
-	podList.Items[4].Status.Phase = corev1.PodFailed
-	err = fakeClient.Update(ctx, &podList.Items[3])
-	assert.Nil(t, err, "Fail to get update pod status")
-	err = fakeClient.Update(ctx, &podList.Items[4])
-	assert.Nil(t, err, "Fail to get update pod status")
-
```

Comment:
I removed this because the test is for the Autoscaler (`WorkersToDelete`). Without this PR, Pod3 and Pod4, which have the status "Failed," would be deleted by `WorkersToDelete`. However, with this PR, they will be removed, but not by `WorkersToDelete`. Maybe I need to remove "PodCrash" from the test name.

Reply:
Ah makes sense